### PR TITLE
Show tf32 info only on rank 0

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

The warning introduced in #16037 is currently shown on every rank. We can assume that the machine has all the same type of GPUs. Otherwise, there would anyway be a significant synchronization overhead. Thus, logging on rank 0 should be sufficient. 

cc @borda @justusschock @awaelchli @carmocca 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

File: src/lightning_lite/accelerators/cuda.py [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
 # limitations under the License.
-import logging
 import os
 import warnings
 from contextlib import contextmanager
 from functools import lru_cache
 from typing import Dict, Generator, List, Optional, Set, Union
 
 import torch
+from lightning_utilities.core.rank_zero import rank_zero_info
```
Comment (carmocca) [inline src/lightning_lite/accelerators/cuda.py:21]: We want to use our import, which will set the correct rank

```suggestion
from lightning_lite.utilities.rank_zero import rank_zero_info
```

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by awaelchli** [comment]
The caplog issues again. Test in CI behaves differently than local. I'm not going to spend more time on this :)
